packages/cozy-pouch-link/src/PouchManager.js

```diff
@@ -0,0 +1,95 @@
+/** Handles the lifecycle of several pouches */
+
+import PouchDB from 'pouchdb'
+import fromPairs from 'lodash/fromPairs'
+import map from 'lodash/map'
+import repeatWithDelay from './repeat'
+
+const DEFAULT_DELAY = 30 * 1000
+
+/* Create a cancellable promise for replication with default options */
+const startReplication = (pouch, url) => {
+  let replication
+  const promise = new Promise((resolve, reject) => {
+    replication = pouch.replicate.from(url, {
+      batch_size: 1000 // we have mostly small documents
+    })
+    replication.on('error', reject).on('complete', resolve)
+  })
+
+  const cancel = () => {
+    if (replication) {
+      replication.cancel()
+    }
+  }
+
+  promise.cancel = cancel
+  return promise
+}
+
+/**
+ * - Creates/Destroys the pouches
+ * - Replicates periodically
+ */
+export default class PouchManager {
+  constructor(doctypes, options) {
+    this.pouches = fromPairs(
+      doctypes.map(doctype => [doctype, new PouchDB(doctype)])
+    )
+    this.options = options
+    this.getReplicationURL = options.getReplicationURL
+  }
+
+  destroy() {
+    this.stopReplicationLoop()
+    map(this.pouches, pouch => pouch.destroy())
```

Comment:
Here we can use `Object.values` (that is [well supported by browsers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values), expect from IE):

```js
Object.values(this.pouches).forEach(pouch => pouch.destroy())
```

The function `map` is often use to return an array. And the anonymous arrow function `pouch => pouch.destroy()` returns the result of `pouch.destroy()`. When reading this whole line, a confusion can be made: we build an array of result of `pouch.destroy()`, but this value is not stored in a variable. Is that something missing?